metalearners/grid_search.py

```diff
@@ -0,0 +1,300 @@
+# Copyright (c) QuantCo 2024-2024
+# SPDX-License-Identifier: BSD-3-Clause
+
+import time
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from typing import Any
+
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.model_selection import ParameterGrid
+
+from metalearners._typing import Matrix, OosMethod, Scoring, Vector, _ScikitModel
+from metalearners.cross_fit_estimator import OVERALL
+from metalearners.metalearner import PROPENSITY_MODEL, MetaLearner
+
+
+@dataclass(frozen=True)
+class _FitAndScoreJob:
+    metalearner: MetaLearner
+    X_train: Matrix
+    y_train: Vector
+    w_train: Vector
+    X_test: Matrix | None
+    y_test: Vector | None
+    w_test: Vector | None
+    oos_method: OosMethod
+    scoring: Scoring | None
+    kwargs: dict
```

Comment:
I think this can cause some confusion because inside this `kwargs` there can be a `fit_params` (basically there can be the following elements: `n_jobs_cross_fitting`, `fit_params`, `synchronize_cross_fitting`, `n_jobs_base_learners`). I would be ok with changing it but I wanted to know if you considered this.

Reply:
I was indeed not aware of this :) What about calling it `metalerner_fit_params` (in contrast to the `(base_learner)fit_params` and adding a brief comment outlining what you just explained?

Reply:
[29db2bb](https://github.com/Quantco/metalearners/pull/9/commits/29db2bb7c54e9207077b54fdc55a92c78c81467f)